bin/critterget.py

```diff
@@ -164,6 +174,181 @@ def apicall_with_response_code(uri, attribs=None):
         return response.status_code, None
 
 
+def time_floor(last_run_time):
+    """
+    Rounds minutes of the last run time down to the nearest ten,
+    and strips the seconds.
+    e.g. 2017-02-14 19:52:02 becomes 2017-02-14 19:50:00
+
+    :param last_run_time: datetime object, the last time the connector ran
+    :return: datetime object
+    """
+    seconds = last_run_time.second
+    minutes = last_run_time.minute % 10
+
+    floored_time = last_run_time - datetime.timedelta(minutes=minutes,
+                                                      seconds=seconds)
+
+    return floored_time
+
+
+def splunk_api_call(uri, method, session_key):
+    """
+    Connect to the local Splunk API
+
+    :param uri: string, Splunk API endpoint
+    :param method: string, either GET or POST
+    :param session_key: string, auth token for Splunk
+    :return: a requests object
+    """
+    url = 'https://localhost:8089{}'.format(uri)
+    headers = {'Authorization': 'Splunk {}'.format(session_key)}
+    params = {OUTPUT_MODE: JSON}
+    splunk_response = None
+
+    if method == GET:
+        try:
+            splunk_response = requests.get(url,
+                                           params=params,
+                                           headers=headers,
+                                           verify=False)
+        except requests.exceptions.Timeout as error:
+            print 'Connection timeout. Splunk API returned an error code:', error
+        except requests.exceptions.ConnectionError as error:
+            print 'Connection error. Splunk API returned an error code:', error
+        except requests.exceptions.HTTPError as error:
+            print 'HTTP error. Splunk API returned an error code:', error
+        except requests.exceptions.RequestException as error:
+            print 'Splunk API retuned an error code:', error
+
+    elif method == POST:
+        try:
+            splunk_response = requests.post(url,
+                                            params=params,
+                                            headers=headers,
+                                            verify=False)
+        except requests.exceptions.Timeout as error:
```

Comment:
Looks like you're repeating your exception handling from the GET implementation.  Can you wrap both methods and just do this once?